pkg/model/keycloak_deployment_test.go

```diff
@@ -489,20 +489,22 @@ func testServiceAccountSet(t *testing.T, deploymentFunction createDeploymentStat
 	assert.Equal(t, "test", serviceAccountName)
 }
 
-func testServiceAccountDefault(t *testing.T, deploymentFunction createDeploymentStatefulSet) {
+func testServiceAccountReconciledSet(t *testing.T, deploymentFunction createDeploymentStatefulSet, reconciliationFunction reconciledDeployment) {
 	//given
 	dbSecret := &v1.Secret{}
 	cr := &v1alpha1.Keycloak{}
-
-	//If serviceAccountName is not set in the cr, then the serviceAccountName should be default
+	statefulSet := deploymentFunction(cr, dbSecret, nil)
 
 	//when
-	serviceAccountName := deploymentFunction(cr, dbSecret, nil).Spec.Template.Spec.ServiceAccountName
 
-	assert.Equal(t, "default", serviceAccountName)
+	//If serviceAccountName is set in the cr, is should manifest itself in the statefulset
+	cr.Spec.KeycloakDeploymentSpec.Experimental.ServiceAccountName = "test2"
```

Comment:
do we have e2e tests like for the new operator for this one? Would be interesting to know what happens when no ServiceAccountName is set (==nil or empty string i guess). I have too little context to evaluate the side effects, tbh, thats why I ask.

Reply:
In any case, the default (no `serviceAccountName` set) is actually tested by all E2E tests as we don't set it anywhere. :)

Reply:
Thanks for the explanation. In that case I approve :)